src/core/src/op/bitwise_right_shift.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "openvino/op/bitwise_right_shift.hpp"
+
+#include "itt.hpp"
+#include "openvino/op/op.hpp"
+#include "openvino/op/util/elementwise_args.hpp"
+#include "openvino/reference/bitwise_right_shift.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v15 {
+namespace right_shift {
+struct Evaluate : ov::element::NoAction<bool> {
+    using ov::element::NoAction<bool>::visit;
+
+    template <element::Type_t ET>
+    static result_type visit(const Tensor& in0, const Tensor& in1, Tensor& out) {
+        using T = typename element_type_traits<ET>::value_type;
+        reference::bitwise_right_shift(in0.data<const T>(),
+                                       in1.data<const T>(),
+                                       out.data<T>(),
+                                       in0.get_shape(),
+                                       in1.get_shape());
+        return true;
+    }
+};
+}  // namespace right_shift
+
+BitwiseRightShift::BitwiseRightShift(const Output<Node>& arg0,
+                                     const Output<Node>& arg1,
+                                     const AutoBroadcastSpec& auto_broadcast)
+    : BinaryElementwiseBitwise(arg0, arg1, auto_broadcast) {
+    constructor_validate_and_infer_types();
+}
+
+std::shared_ptr<Node> BitwiseRightShift::clone_with_new_inputs(const OutputVector& new_args) const {
+    OV_OP_SCOPE(v15_BitwiseRightShift_clone_with_new_inputs);
+    check_new_args_count(this, new_args);
+    return std::make_shared<BitwiseRightShift>(new_args[0], new_args[1], get_autob());
+}
+
+bool BitwiseRightShift::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
+    OV_OP_SCOPE(v15_BitwiseRightShift_evaluate);
+    OPENVINO_ASSERT(outputs.size() == 1);
+    outputs[0].set_shape(infer_broadcast_shape(this, inputs));
+    using namespace ov::element;
+    return IF_TYPE_OF(v15_BitwiseRightShift_evaluate,
```

Comment:
Just to double check if this macro should not be used `IF_TYPE_OF_CONVERT_TENSORS` but as floating points are not supported I think currently used macro is fine.

Reply:
`IF_TYPE_OF_CONVERT_TENSORS` calls `evaluate_node_with_unsupported_precision` and this path is not tested by template tests. In fact any type conversion for bit shifts can lead to unexpected results, so in my opinion it's better to keep simple `IF_TYPE_OF`.
https://github.com/openvinotoolkit/openvino/blob/dcd434214635ad444de4512588a60a65023f38c6/src/core/src/constant_fold_utils.cpp#L246-L257